vhost/src/vhost_user/message.rs

```diff
@@ -777,12 +782,28 @@ impl VhostUserMsgValidator for VhostUserVringAddr {
     fn is_valid(&self) -> bool {
         if (self.flags & !VhostUserVringAddrFlags::all().bits()) != 0 {
             return false;
-        } else if self.descriptor & 0xf != 0 {
-            return false;
-        } else if self.available & 0x1 != 0 {
-            return false;
-        } else if self.used & 0x3 != 0 {
-            return false;
+        }
+
+        // Check if packed ring format
+        if self.flags & VhostUserVringAddrFlags::VHOST_VRING_F_PACKED.bits() != 0 {
+            // Packed ring validation:
+            // - descriptor ring must be 16-byte aligned
+            if self.descriptor & 0xf != 0 {
+                return false;
+            }
+            // - driver/device event suppression areas must be 4-byte aligned
+            if self.available & 0x3 != 0 || self.used & 0x3 != 0 {
+                return false;
+            }
+        } else {
+            // Split ring validation (current logic)
+            if self.descriptor & 0xf != 0 {
+                return false;
+            } else if self.available & 0x1 != 0 {
+                return false;
+            } else if self.used & 0x3 != 0 {
+                return false;
+            }
```

Comment:
this is almost duplicated code, the only difference is the available's alignment, could this be simplified?